client/app/lib/defer.js

```diff
@@ -0,0 +1,8 @@
+export default function defer() {
```

Comment:
Query service is the only place where this is needed. I'd just add this function there